viz_scripts/survey_responses.ipynb

```diff
@@ -0,0 +1,343 @@
+    "    for i in range(len(labels)):\n",
+    "        l_labels = str(labels[i]).split(\" \")\n",
+    "        for k in range(len(l_labels)):\n",
+    "            l_labels[k] = opt_dict[l_labels[k]]\n",
+    "        sep = \"\\n\"\n",
+    "        labels[i] = sep.join(l_labels)\n",
+    "    \n",
+    "    return labels\n",
+    "\n",
+    "#create a debug dataframe\n",
+    "def generate_debug_df(program, include_test_users, full_df, labeled_df):\n",
+    "    debug_df = pd.DataFrame.from_dict({\n",
+    "            \"year\": year,\n",
+    "            \"month\": month,\n",
+    "            \"Registered_participants\": len(scaffolding.get_participant_uuids(program, include_test_users)),\n",
+    "            \"Participants_with_at_least_one_trip\": scaffolding.unique_users(full_df),\n",
+    "            \"Participant_with_at_least_one_labeled_trip\": scaffolding.unique_users(labeled_df),\n",
+    "            \"Trips_with_at_least_one_label\": len(labeled_df)\n",
+    "            },\n",
+    "        orient='index', columns=[\"value\"])\n",
+    "    \n",
+    "    return debug_df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5319980e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#list of all surveys that are not a \"UserProfileSurvey\"\n",
+    "survey_list = list(survey_info['surveys'].keys())\n",
+    "survey_list.remove('UserProfileSurvey')\n",
+    "sheet_list = get_sheet_links(survey_list)\n",
+    "\n",
+    "print('survey sheets: ', sheet_list)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3b355efd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#load all of the composite trips - require sections\n",
+    "tq = scaffolding.get_time_query(year, month)\n",
+    "all_composite_trips = scaffolding.load_all_composite_trips(tq)\n",
+    "#we need to filter out trips (based on if including test users)\n",
+    "all_composite_trips = scaffolding.filter_composite_trips(all_composite_trips, program, include_test_users)\n",
+    "\n",
+    "if len(all_composite_trips) > 0:\n",
+    "    #remove blank inputs\n",
+    "    survey_trips = all_composite_trips[all_composite_trips['user_input'] != {}]\n",
+    "\n",
+    "    #survey counts df\n",
+    "    survey_trips = survey_trips.reset_index()\n",
+    "    survey_trips['survey_name'] = survey_trips.user_input.apply(lambda sr: sr['trip_user_input']['data']['name'])\n",
+    "\n",
+    "    #gather the cols needed for charts and text\n",
+    "    survey_trips = survey_trips[['survey_name', 'user_id', 'user_input']]\n",
+    "    \n",
+    "    # transform data to meet a data format requirement\n",
+    "    all_composite_trips['sections'] = all_composite_trips['sections'].apply(lambda x: [x[0]['data']] if len(x) > 0 else x)\n",
+    "\n",
+    "    #only want trips with sections\n",
+    "    all_composite_trips = all_composite_trips[all_composite_trips['sections'].str.len() > 0]\n",
```

Comment:
what is the data format requirement that requires us to have sections?

Reply:
it was the old way we filtered what survey was assigned to a trip

Reply:
I see that, we should switch to confirmedMode as above. This is probably why none of the plots were working, none of them matched the current conditional in the config.

Reply:
addressed in [548a449](https://github.com/e-mission/em-public-dashboard/pull/124/commits/548a449d6b46130a0441eb39b372adca0ee56459)